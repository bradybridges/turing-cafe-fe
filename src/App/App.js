import React, { Component } from 'react';
import './App.css';
import { fetchReservations } from '../apiCalls';
import ReservationCard from '../ReservationCard/ReservationCard';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      reservations: []
    }
  }

  componentDidMount() {
    fetchReservations()
      .then( reservations => {
        this.setState({ reservations });
        console.log(reservations);
      })
  }


  render() {

    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>

        </div>
        <div className='resy-container'>
          {this.state.reservations.map(res => <ReservationCard name={res.name} date={res.date} time={res.time} number={res.number} id={res.id} />)}
        </div>
      </div>
    )
  }
}

export default App;
