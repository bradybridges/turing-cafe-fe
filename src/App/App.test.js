import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;
  const mockReservations = [
    {date: '12/22', id: 1, name: 'Brady', number: 2, time: '8:00'},
    {date: '12/23', id: 2, name: 'Randy', number: 5, time: '5:00'}
  ];

  beforeEach(() => {
    wrapper = shallow(<App />);
    wrapper.setState({reservations: mockReservations});

  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  
});

