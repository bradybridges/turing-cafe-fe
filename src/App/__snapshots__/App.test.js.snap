// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <ReservationForm
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <ReservationCard
      cancelReservation={[Function]}
      date="12/22"
      id={1}
      key="0"
      name="Brady"
      number={2}
      time="8:00"
    />
    <ReservationCard
      cancelReservation={[Function]}
      date="12/23"
      id={2}
      key="1"
      name="Randy"
      number={5}
      time="5:00"
    />
  </div>
</div>
`;
